scrapy/statscollectors.py

```diff
@@ -43,13 +43,16 @@ def open_spider(self, spider):
 
     def close_spider(self, spider, reason):
         if self._dump:
-            logger.info("Dumping Scrapy stats:\n" + pprint.pformat(self._stats),
+            logger.info("Dumping Scrapy stats:\n{}".format(self),
                         extra={'spider': spider})
         self._persist_stats(self._stats, spider)
 
     def _persist_stats(self, stats, spider):
         pass
 
+    def __format__(self, _):
```

Comment:
I think the way to go is to check the format, use pformat if a string is requested, use repr if repr is requested, raise for numeric formats. But it looks like a lot of code without a clear purpose; maybe using `__str__` hook instead of `__format__` is easier.